test/controllers/user.controller.test.js

```diff
@@ -1857,6 +1857,201 @@ describe('User Controller', () => {
         });
       });
     });
+
+    describe('Delete director or signatories', () => {
+      let vendorToken;
+      const signatoryId = mongoose.Types.ObjectId();
+      const nonSignatoryId = mongoose.Types.ObjectId();
+      const vendorUser = UserFactory.build(
+        {
+          role: USER_ROLE.VENDOR,
+          activated: true,
+          vendor: {
+            companyName: 'Highrachy Investment Limited',
+            directors: [
+              {
+                _id: nonSignatoryId,
+                name: 'Jane Doe',
+                isSignatory: false,
+                phone: '08012345678',
+              },
+              {
+                _id: signatoryId,
+                name: 'Samuel',
+                isSignatory: true,
+                signature: 'signature.png',
+                phone: '08012345678',
+              },
+              {
+                name: 'John',
+                isSignatory: false,
+                phone: '08012345678',
+              },
+            ],
+          },
+        },
+        { generateId: true },
+      );
+      const invalidUser = UserFactory.build({ role: USER_ROLE.ADMIN }, { generateId: true });
```

Comment:
It is preferable to use a USER role for an invalid user here. An Admin role for be used for subsequent tests

```suggestion
        const invalidUser = UserFactory.build({ role: USER_ROLE.USER }, { generateId: true });
```